File: C#/uri-1031.cs
// URI - 1031: https://www.urionlinejudge.com.br/judge/pt/problems/view/1031

using System;
using System.Collections.Generic;

namespace URI
{
    class Program
    {
        static void Main(string[] args)
        {
            List<int> regions = new List<int>();
            int numberOfRegions = 1, interval = 0;
            while (numberOfRegions != 0)
            {
                numberOfRegions = int.Parse(Console.ReadLine());
                if (numberOfRegions == 0)
                    break;
                do
                {
                    if (regions.Count > 0)
                        regions.RemoveAt(0);
                    for (int i = 1; i <= numberOfRegions; i++)
                        regions.Add(i);
                    interval++;
                } while (EnergyCut(regions, interval, 0) != 13);
                Console.WriteLine(interval);
                interval = 0;
            }
        }

        private static int EnergyCut(List<int> regions, int interval, int cut)
        {
            while (regions.Count > 1)
            {
                if (cut > regions.Count - 1)
                    cut -= regions.Count;
                if (cut == regions.Count)
                    cut = 0;
                if (cut <= regions.Count)
                {
                    regions.RemoveAt(cut);
                    cut += interval - 1;
                }
            }
            return regions[0];
        }
    }
}

File: JavaScript/uri-1031.js
// URI - 1031: https://www.urionlinejudge.com.br/judge/pt/problems/view/1031

const input = require('fs').readFileSync('/dev/stdin', 'utf8');
const lines = input.split('\n');

let numberOfRegions, regions = [], interval = 0;

while (true) {
    numberOfRegions = lines.shift();
    if (numberOfRegions == 0)
        break;
    
    for (let i = 1; i <= numberOfRegions; i++)
        regions.push(i);

    do {
        interval++;
    } while (energyCut(regions.slice(0), 0, interval) != 13);
    
    console.log(interval);
    interval = 0;
    regions = [];
}

function energyCut(regions, cut, interval) {
    while (regions.length > 1) {
        if (cut > regions.length - 1)
            cut -= regions.length;

        if (cut == regions.length)
            cut = 0;

        if (cut <= regions.length - 1) {
            regions.splice(cut, 1);
            cut += interval - 1;
        }
    }
    return regions[0];
}